clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1580,10 +1580,18 @@ protected void checkIngestionProgress(Store store) throws InterruptedException {
   }
 
   protected void updateIngestionRoleIfStoreChanged(Store store) throws InterruptedException {
+    Version version = store.getVersion(versionNumber);
+    if (version == null) {
+      LOGGER.warn(
+          "Store version: {} is not found in store: {}. Skipping resubscription check.",
+          versionNumber,
+          storeName);
+      return;
+    }
     PartitionReplicaIngestionContext.VersionRole newVersionRole =
-        PartitionReplicaIngestionContext.getStoreVersionRole(versionTopic, store);
+        PartitionReplicaIngestionContext.getStoreVersionRole(version, store);
     PartitionReplicaIngestionContext.WorkloadType newWorkloadType =
-        PartitionReplicaIngestionContext.getWorkloadType(versionTopic, store);
+        PartitionReplicaIngestionContext.getWorkloadType(version, store);
     if (serverConfig.isResubscriptionTriggeredByVersionIngestionContextChangeEnabled() && isHybridMode()) {
```

Comment:
Let’s move the newVersionRole and newWorkloadType lookups inside the if block to prevent unnecessary memory allocations.